scripts/scripts.js

```diff
@@ -1,195 +1,194 @@
                         <p>${gif.source_tld}</p>
                     </div>
                 </div>
             </figcaption>
         
            `;
-            //4. agregar el nodo html a el padre.
-            main.appendChild(figure);
+    //4. agregar el nodo html a el padre.
+    main.appendChild(figure);
+  });
+};
+/** Updates search termn in view */
+const updateParamView = (q) => {
+  let text = q ? q : 'Todo';
+  let searchElement = document.getElementById('search-term');
+  searchElement.innerHTML = text;
+};
+
+const init = async (offset = 0) => {
+  const searchTerm = getUrlParams();
+  const giphyObj = new GiphyObj();
+  giphyObj.q = searchTerm;
+  updateParamView(searchTerm);
+  getHistory();
+
+  if (searchTerm) {
+    loadGifs(giphyObj.search_api(), updateMainView);
+  } else {
+    loadGifs(giphyObj.trending_api(), updateMainView);
+  }
+
+  // Improved scroll event handling
+  let loading = false; // Prevent multiple scroll events
+
+  window.addEventListener('scroll', () => {
+    if (
+      !loading &&
+      window.innerHeight + window.scrollY >= document.body.offsetHeight - 200
+    ) {
+      loading = true;
+      giphyObj.updateOffset();
+
+      if (searchTerm) {
+        loadGifs(giphyObj.search_api(), (gifs) => {
+          updateMainView(gifs);
+          loading = false;
         });
-
+      } else {
+        loadGifs(giphyObj.trending_api(), (gifs) => {
+          updateMainView(gifs);
+          loading = false;
+        });
+      }
     }
-    /** Updates search termn in view */
-const updateParamView = (q) => {
-    let text = q ? q : "Todo";
-    let searchElement = document.getElementById("search-term")
-    searchElement.innerHTML = text;
-}
-
-
-
-const init = async(offset = 0) => {
-        // check param from url
-        const searchTerm = getUrlParams();
-        let giphyObj = new GiphyObj();
-        giphyObj.q = searchTerm;
-        updateParamView(searchTerm);
-
-        //lod history
-        getHistory();
+  });
 
+  const searchBar = document.getElementById('search');
+  searchBar.addEventListener('search', (e) => {
+    searchEvent(e, giphyObj);
+  });
 
+  document.getElementById('search-btn').addEventListener('click', (e) => {
+    searchEvent(e, giphyObj);
+  });
+};
 
-        if (searchTerm) {
-            loadGifs(giphyObj.search_api(), updateMainView);
-        } else {
-            loadGifs(giphyObj.trending_api(), updateMainView);
-        }
-
-        //Inifinite loading wth scroll event
-        window.addEventListener('scroll', () => {
-            console.log("scrolled", window.scrollY) //scrolled from top
-            console.log(window.innerHeight) //visible part of screen
-
-            if (window.scrollY + window.innerHeight >= document.documentElement.scrollHeight) {
-                giphyObj.updateOffset();
-                //update css
-                let list = document.querySelector("#gifs-list");
-                list.style.height = `${giphyObj.offset*10}vh`;
```

Comment:
Keeping the height of the list item in infinite scrolling is needed. It allows the page to dynamically extend its visible area, accommodating more content as users scroll, preventing layout issues. Without it, new content may overlap or appear off-screen, causing display problems.
Like this
[screen-capture (1).webm](https://github.com/lcrojano/Giphy_Explorer/assets/5177737/a35f0fcf-2fb4-4823-b10d-43551894d2f1)